src/rimage.c

```diff
@@ -122,8 +122,8 @@ int main(int argc, char *argv[])
 		return -EINVAL;
 	}
 
-	/* make sure we have an outfile if not verifying */
-	if ((!image.out_file && !image.verify_file)) {
+	/* We are either signing or verifying an image, not both or neither */
+	if (!(!!image.out_file ^ !!image.verify_file)) {
```

Comment:
possible to figure out a more easier expression for better understanding?

Reply:
" not both or neither " would this not imply to fail if ```!out_file == !verify_file``` ? Pls check my logic here though.